File: 프로그래머스/lv2/42587. 프로세스/프로세스.java
import java.util.PriorityQueue;
import java.util.Collections;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        PriorityQueue<Pair> pr = new PriorityQueue<Pair>();
        
        for (int i = 0; i < priorities.length; i++) {
            pr.add(new Pair(priorities[i], i));
        }
        int t = 0;
        int cnt = 1;
        while (!pr.isEmpty()) {
            int v = pr.poll().value;
            while (v != priorities[t % priorities.length]) {
                t++;
            }
            if (t % priorities.length == location) break;
            else {
                cnt++;
                t++;
            }
        }
        
        return cnt;
    }
}

class Pair implements Comparable<Pair> {
    int value;
    int index;
    
    public Pair(int value, int index) {
        this.value = value;
        this.index = index;
    }
    
    @Override
    public int compareTo(Pair other) {
        return Integer.compare(-this.value, -other.value);
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    q = deque()
    n = len(priorities)
    i = 0
    v = [0] * n
    while sum(v) < n:
        if not v[i%n]:
            if q: 
                while q and (q[-1][0] < priorities[i%n]):
                    v[q.pop()[1]] = 0
            q.append([priorities[i%n], i%n])
            v[i%n] = 1
        i += 1
    
    for j in range(n):
        if q[j][1] == location:
            return j + 1
    return 0